src/main/cpp/TrajMath.cpp

```diff
@@ -0,0 +1,72 @@
+#include "TrajMath.h"
+
+t34::TrajMath::TrajMath
+(
+    double note_max_velocity_mps,
+    double target_height_meters,
+    double limelight_height_meters,
+    double shooter_angle,
+    double limelight_angle
+):
+    m_note_max_velocity_mps(note_max_velocity_mps),
+    m_target_height_meters(target_height_meters),
+    m_limelight_height_meters(limelight_height_meters),
+    m_shooter_angle(shooter_angle),
+    m_limelight_angle(limelight_angle),
+    m_target_tx(0.0),
+    m_target_ty(0.0)
+{}     
+
+double t34::TrajMath::GetFiringAngleDeg()
+{
+    //                   ____________________
+    //          ⎛  v² - √ v⁴ - g(gx² + 2v²y)   ⎞
+    // θ = atan ⎜ ———————————————————————————— ⎟
+    //          ⎝               gx             ⎠
+    //
+    // Source:
+    //   Solving Ballistic Trajectories <https://www.forrestthewoods.com/blog/solving_ballistic_trajectories/>
+    //   See section “Firing Angle to Hit Stationary Target.”
+
+    const auto v2 = m_note_max_velocity_mps * m_note_max_velocity_mps;
+    const auto v4 = v2 * v2;
+    const auto x  = m_target_distance_meters;
+    const auto x2 = x * x;
+    const auto y  = m_target_height_meters;
+
+    const auto gx  = g * x;
+    const auto gx2 = g * x2;
+    const auto v2y = v2 * y;
+
+    const auto numerator = v2 - sqrt(v4 - (g * (gx2 + (2*v2y))));
+    const auto denominator = gx;
+    const auto θ = atan(numerator / denominator);
+
+    return RAD_TO_DEG(θ);
+}
+
+bool t34::TrajMath::IsInRange()
+{
+    double rate = (
+        tan(m_shooter_angle) - 
+        (g * m_target_distance_meters) / (pow(m_shooter_angle, 2) * pow(m_note_max_velocity_mps, 2))
+    );
+
+    if (rate > 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+double t34::TrajMath::GetDistanceFromTarget()
+{
+    return (
+        (m_target_height_meters - m_limelight_height_meters) / tan(
+            RAD_TO_DEG( (m_limelight_angle + m_target_ty) )
+        )
```

Comment:
I think we have it backward here. `tan` [takes radians](https://en.cppreference.com/w/c/numeric/math/tan), not degrees:

```suggestion
        (m_target_height_meters - m_limelight_height_meters) / tan(
            DEG_TO_RAD( (m_limelight_angle + m_target_ty) )
        )
```

That's assuming, by the way, that `m_limelight_angle` and `m_target_ty` are in degrees. If that's correct, it would probably be a good idea to note it in the variable names:

```suggestion
        (m_target_height_meters - m_limelight_height_meters) / tan(
            DEG_TO_RAD( (m_limelight_angle_deg + m_target_ty_deg) )
        )
```

Then again, looking at the name `m_target_ty`, I'm guessing it's actually a lateral distance, not an angle. Are we sure we have the right formula here?